artifactory/services/utils/replication.go

```diff
@@ -32,13 +36,35 @@ type ReplicationParams struct {
 	IncludePathPrefixPattern string
 }
 
+// UnmarshalJSON overrides the default JSON unmarshal function because the POST request to create a replication
+// has a field named `proxy` but the GET request returns a JSON with a field named `proxyRef`
+func (rp *ReplicationParams) UnmarshalJSON(data []byte) error {
```

Comment:
I suggest taking a different approach instead of defining the ```UnmarshalJSON``` function.
We can do the following:
1. Remove the ```Proxy``` property from the ```ReplicationBody``` struct.
2. Create two new structs named ```GetReplicationBody``` and ```UpdateReplicationBody```.
3. The two structs should reference the ```ReplicationBody``` struct, and only add the ```Proxy``` or ```ProxyRef``` properties.

This way, the fields are replicated na the ```ReplicationBody``` struct is used as a base for ```GetReplicationBody``` and ```UpdateReplicationBody```.

Reply:
Thank you for the suggestion!  I have made the changes accordingly.

Reply:
@eyalbe4, just a curious consumer here. Would you mind explaining why you prefer your approach to the original proposed change? Seems like the initial change would have been backwards compatible. I see your later comment about being generally okay with backwards incompatible changes, but if there is a way to do something in a non-breaking way, why not take that approach? Just trying to understand rationale here, not trying to be a pain!